File: app/src/main/java/co/tpcreative/supersafe/model/DriveEvent.java
package co.tpcreative.supersafe.model;
import com.google.gson.Gson;
import java.io.Serializable;
import co.tpcreative.supersafe.common.util.Utils;

public class DriveEvent implements Serializable {

    private static DriveEvent instance;
    public String items_id;
    public int fileType;
    private static final String TAG = DriveEvent.class.getSimpleName();


    public static DriveEvent getInstance(){
        if (instance==null){
            instance = new DriveEvent();
        }
        return instance;
    }

    public String convertToHex(final String value){
        try {
            return Utils.stringToHex(value);
        }
        catch (Exception e){
            Utils.Log(TAG,"Error :" + e.getMessage());
            Utils.onWriteLog(TAG +"-" + e.getMessage(),EnumStatus.WRITE_FILE);
            e.printStackTrace();
        }
        return null;
    }

    public DriveEvent hexToObject(String value){
        try {
            String result = Utils.hexToString(value);
            return new Gson().fromJson(result,DriveEvent.class);
        }
        catch (Exception e){
            Utils.Log(TAG,"Error :" + e.getMessage());
            Utils.onWriteLog(TAG +"-" + e.getMessage(),EnumStatus.WRITE_FILE);
            e.printStackTrace();
        }
        return null;
    }




}

File: app/src/main/java/co/tpcreative/supersafe/model/DriveEvent.kt
package co.tpcreative.supersafe.model
import co.tpcreative.supersafe.common.util.Utils
import com.google.gson.Gson
import java.io.Serializable

class DriveEvent : Serializable {
    var items_id: String? = null
    var fileType = 0
    fun convertToHex(value: String?): String? {
        try {
            return Utils.stringToHex(value)
        } catch (e: Exception) {
            Utils.Log(DriveEvent.Companion.TAG, "Error :" + e.message)
            Utils.onWriteLog(DriveEvent.Companion.TAG + "-" + e.message, EnumStatus.WRITE_FILE)
            e.printStackTrace()
        }
        return null
    }

    fun hexToObject(value: String?): DriveEvent? {
        try {
            val result: String? = Utils.hexToString(value)
            return Gson().fromJson(result, DriveEvent::class.java)
        } catch (e: Exception) {
            Utils.Log(DriveEvent.Companion.TAG, "Error :" + e.message)
            Utils.onWriteLog(DriveEvent.Companion.TAG + "-" + e.message, EnumStatus.WRITE_FILE)
            e.printStackTrace()
        }
        return null
    }

    companion object {
        private var instance: DriveEvent? = null
        private val TAG = DriveEvent::class.java.simpleName
        fun getInstance(): DriveEvent? {
            if (instance == null) {
                instance = DriveEvent()
            }
            return instance
        }
    }
}